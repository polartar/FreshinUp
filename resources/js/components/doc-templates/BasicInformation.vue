<template>
  <v-layout>
    <v-flex
      xs12
      sm12
      md8
      xl9
    >
      <v-card>
        <v-card-title>
          <h3 class="grey--text">
            Basic Information
          </h3>
          <v-progress-linear
            v-if="isLoading"
            indeterminate
          />
        </v-card-title>
        <v-divider />
        <v-card-text>
          <div>
            <v-flex xs12>
              <div class="mb-2 text-uppercase grey--text font-weight-bold">
                Title
              </div>
              <v-text-field
                v-model="title"
                placeholder="Title"
                single-line
                outline
              />
            </v-flex>
            <v-flex xs12>
              <div class="mb-2 text-uppercase grey--text font-weight-bold">
                Short description
              </div>
              <v-text-field
                v-model="description"
                placeholder="Short description"
                single-line
                outline
              />
            </v-flex>
            <v-flex xs12>
              <div class="mb-2 text-uppercase grey--text font-weight-bold">
                Content
              </div>
              <v-textarea
                v-model="content"
                placeholder="Short description"
                single-line
                outline
              />
            </v-flex>
          </div>
        </v-card-text>
        <v-card-actions>
          <v-btn
            @click="cancel"
          >
            Cancel
          </v-btn>
          <v-btn
            color="primary"
            :loading="isLoading"
            @click="save"
          >
            {{ isNew? 'Submit' : 'Save changes' }}
          </v-btn>
          <v-spacer />
          <v-btn
            @click="deleteItem"
          >
            Delete
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-flex>
    <v-flex
      xs12
      sm12
      md4
      xl3
      class="ml-2"
    >
      <v-card>
        <v-card-text>
          <h3
            v-if="!isNew"
            class="font-weight-bold mb-2"
          >
            Last Modified on
          </h3>
          <span
            v-if="!isNew"
            class="mb-2"
          >{{ formatDate(updated_at) }} by <u>{{ updaterName }}</u></span>
          <v-select
            v-model="status_id"
            :items="statuses"
            item-value="id"
            item-text="name"
            outline
            single-line
          />
        </v-card-text>
        <v-card-actions>
          <v-layout
            text-xs-center
            row
            wrap
          >
            <v-flex
              xs12
              class="mb-2"
            >
              <v-btn
                color="primary"
                :loading="isLoading"
                @click="save"
              >
                {{ isNew? 'Submit' : 'Save changes' }}
              </v-btn>
            </v-flex>
            <v-flex
              xs12
              class="mb-2"
            >
              <v-btn
                disabled
              >
                Preview
              </v-btn>
            </v-flex>
            <v-flex xs12>
              <a
                v-if="!isNew"
                href="#/move-to-trash"
                @click.prevent="deleteItem"
              >Move to trash</a>
            </v-flex>
          </v-layout>
        </v-card-actions>
      </v-card>
    </v-flex>
  </v-layout>
</template>

<script>
import MapValueKeysToData from '~/mixins/MapValueKeysToData'
import get from 'lodash/get'
import FormatDate from '@freshinup/core-ui/src/mixins/FormatDate'

export const DEFAULT_TEMPLATE = {
  id: '',
  uuid: '',
  title: '',
  description: '',
  content: '',
  status_id: 1,
  updated_at: '',
  updated_by: null
}

export const EDITOR_OPTIONS = {
  height: 500,
  menubar: false,
  plugins: [
    'advlist autolink lists link image charmap',
    'searchreplace visualblocks code fullscreen',
    'print preview anchor insertdatetime media',
    'paste code help wordcount table'
  ],
  toolbar:
    ['undo', 'redo', '|', 'formatselect', '|', 'bold',
      'italic', 'backcolor', '|', 'alignleft', 'aligncenter',
      'alignright', 'alignjustify', '|', 'bullist', 'numlist',
      'outdent', 'indent', '|', 'removeformat', '|', 'help'].join(' ')
}
export default {
  mixins: [MapValueKeysToData, FormatDate],
  props: {
    isLoading: { type: Boolean, default: false },
    // Override value from MapValueKeysToData mixin to get the default values
    value: { type: Object, default: () => DEFAULT_TEMPLATE },
    statuses: { type: Object, default: () => [] }
  },
  data () {
    return {
      apiKey: '',
      ...DEFAULT_TEMPLATE
    }
  },
  computed: {
    // TODO useful for https://github.com/FreshinUp/foodfleet/issues/517
    editorOptions () {
      return EDITOR_OPTIONS
    },
    isNew () {
      return !get(this, 'uuid')
    },
    updaterName () {
      return get(this, 'updated_by.name')
    }
  },
  methods: {
    get,
    deleteItem () {
      this.$emit('delete', this.payload)
    },
    cancel () {
      this.$emit('cancel')
    }
  }
}
</script>

<style scoped>

</style>
