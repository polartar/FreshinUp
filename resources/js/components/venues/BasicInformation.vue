<template>
  <v-card>
    <v-card-title>
      Basic Information
      <v-progress-linear
        v-if="loading"
        indeterminate
      />
    </v-card-title>
    <v-divider />
    <v-layout pa-3>
      <v-flex
        xs8
        pr-3
      >
        <v-flex
          xs12
        >
          <div class="mb-2 text-uppercase grey--text font-weight-bold">
            Venue name
          </div>
          <v-text-field
            v-model="name"
            placeholder="Name"
            single-line
            outline
          />
        </v-flex>
        <v-flex
          xs12
        >
          <div class="mb-2 text-uppercase grey--text font-weight-bold">
            Address line 1
          </div>
          <v-text-field
            v-model="address_line_1"
            placeholder="Address 1"
            single-line
            outline
          />
        </v-flex>
        <v-flex
          xs12
        >
          <div class="mb-2 text-uppercase grey--text font-weight-bold">
            Address line 2
          </div>
          <v-text-field
            v-model="address_line_2"
            placeholder="Address 2"
            single-line
            outline
          />
        </v-flex>
        <v-layout row>
          <v-flex xs8>
            <div class="mb-2 text-uppercase grey--text font-weight-bold">
              Owned by
            </div>
            <v-flex
              xs12
              sm6
              md8
              align-center
              layout
            >
              <v-avatar
                :tile="false"
                :size="80"
                color="grey lighten-4"
              >
                <img
                  :src="owner.avatar"
                  alt="avatar"
                >
              </v-avatar>
              <div class="mx-2 px-2">
                <div class="primary--text subheading">
                  {{ owner.name }}
                </div>
                <div class="grey--text text--darken-2">
                  {{ owner.email }}
                </div>
                <div class="grey--text text--darken-2">
                  {{ owner.mobile_phone }}
                </div>
              </div>
            </v-flex>
          </v-flex>
          <v-flex
            xs4
            layout
            column
            align-content-center
            justify-center
          >
            <v-btn
              depressed
              disabled
              color="primary"
            >
              Change user
            </v-btn>
          </v-flex>
        </v-layout>
      </v-flex>
      <v-flex
        xs4
        pl-3
      >
        MAP
      </v-flex>
    </v-layout>
    <v-divider />
    <v-card-actions class="pa-3 d-flex justify-space-between">
      <div class="d-flex">
        <div>
          <v-btn
            depressed
            disabled
            @click="onCancel"
          >
            Cancel
          </v-btn>
          <v-btn
            depressed
            disabled
            color="primary"
            @click="save"
          >
            Save Changes
          </v-btn>
        </div>
        <div class="text-xs-right">
          <v-btn
            depressed
            disabled
            @click="onDeleteVenue"
          >
            Delete Venue
          </v-btn>
        </div>
      </div>
    </v-card-actions>
  </v-card>
</template>
<script>
import MapValueKeysToData from '../../mixins/MapValueKeysToData'
import pick from 'lodash/pick'
import keys from 'lodash/keys'

export const DEFAULT_VENUE = {
  uuid: '',
  name: '',
  address_line_1: '',
  address_line_2: '',
  owner_uuid: '',
  owner: {
    name: '',
    email: '',
    mobile_phone: '',
    avatar: ''
  }
}

export default {
  mixins: [MapValueKeysToData],
  props: {
    loading: { type: Boolean, default: false }
  },
  data () {
    return {
      ...DEFAULT_VENUE
    }
  },
  methods: {
    onCancel () {
      this.$emit('cancel')
    },
    onDeleteVenue () {
      this.$emit('delete', pick(this, keys(this.value)))
    }
  }
}
</script>
