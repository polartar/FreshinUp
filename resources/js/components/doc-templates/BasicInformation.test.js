import * as Stories from './BasicInformation.stories'
import Component, {EDITOR_OPTIONS} from './BasicInformation'
import { shallowMount } from '@vue/test-utils'
import { FIXTURE_USERS } from '../../../../tests/Javascript/__data__/users'
import { FIXTURE_DOCUMENT_TEMPLATES } from '../../../../tests/Javascript/__data__/documentTemplates'
import { FIXTURE_DOCUMENT_TEMPLATE_STATUSES } from '../../../../tests/Javascript/__data__/documentTemplateStatuses'
describe('component/doc-templates/BasicInformation', () => {
  describe('Snapshots', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Loading', async () => {
      const wrapper = mount(Stories.Loading())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Populated', async () => {
      const wrapper = mount(Stories.Populated())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props & Computed', () => {
    test('value', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.value).toMatchObject(DEFAULT_TEMPLATE)

      const template = FIXTURE_DOCUMENT_TEMPLATES[0]
      wrapper.setProps({
        value: template
      })
      expect(wrapper.vm.value).toMatchObject(template)
    })
    test('editorOptions', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.editorOptions).toMatchObject(EDITOR_OPTIONS)
    })
    test('isNew', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.isNew).toBe(false)

      wrapper.setProps({
        value: {
          uuid: 'abc123'
        }
      })
      expect(wrapper.vm.isNew).toBe(true)
    })
    test('statuses', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.statuses).toHaveLength(0)

      wrapper.setProps({
        statuses: FIXTURE_DOCUMENT_TEMPLATE_STATUSES
      })
      expect(wrapper.vm.statuses).toMatchObject(FIXTURE_DOCUMENT_TEMPLATE_STATUSES)
    })
    test('updaterName', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.updaterName).toBeNull()

      const user = FIXTURE_USERS[0]
      wrapper.setProps({
        value: {
          updated_by: user
        }
      })
      expect(wrapper.vm.updaterName).toEqual(user.name)
    })
  })

  describe('Methods', () => {
    test('deleteItem()', async () => {
      const template = FIXTURE_DOCUMENT_TEMPLATES[0]
      const wrapper = shallowMount(Component, {
        propsData: {
          value: template
        }
      })
      wrapper.vm.deleteItem()
      const event = wrapper.emitted().delete
      expect(event).toBeTruthy()
      expect(event[0][0]).toMatchObject(template)
    })
  })
})
