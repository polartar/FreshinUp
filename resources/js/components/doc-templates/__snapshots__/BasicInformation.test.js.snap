// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/doc-templates/BasicInformation Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="layout"
  >
    <div
      class="flex xs12 sm12 md8 xl9"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__title"
        >
          <h3
            class="grey--text"
          >
            
          Basic Information
        
          </h3>
           
          <!---->
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <div
          class="v-card__text"
        >
          <div>
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Title
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Title"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Short description
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Short description"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Content
            
              </div>
               
              <div
                class="v-input v-textarea v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <textarea
                        placeholder="Short description"
                        rows="5"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <span
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Cancel
        
            </div>
          </button>
           
          <button
            class="v-btn theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Submit
        
            </div>
          </button>
           
          <div
            class="spacer"
          />
           
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Delete
        
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="flex ml-2 xs12 sm12 md4 xl3"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <!---->
           
          <!---->
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="layout text-xs-center row wrap"
          >
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Submit
            
                </div>
              </button>
            </div>
             
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn v-btn--disabled theme--light"
                disabled="disabled"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Preview
            
                </div>
              </button>
            </div>
             
            <div
              class="flex xs12"
            >
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`component/doc-templates/BasicInformation Snapshots Loading 1`] = `
<div
  class="container"
>
  <div
    class="layout"
  >
    <div
      class="flex xs12 sm12 md8 xl9"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__title"
        >
          <h3
            class="grey--text"
          >
            
          Basic Information
        
          </h3>
           
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            class="v-progress-linear"
            role="progressbar"
            style="height: 7px;"
          >
            <div
              class="v-progress-linear__background primary"
              style="height: 7px; opacity: 0.3; width: 100%;"
            />
            <div
              class="v-progress-linear__bar"
            >
              <div
                class="v-progress-linear__bar__indeterminate v-progress-linear__bar__indeterminate--active"
              >
                <div
                  class="v-progress-linear__bar__indeterminate long primary"
                />
                <div
                  class="v-progress-linear__bar__indeterminate short primary"
                />
              </div>
              <!---->
            </div>
          </div>
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <div
          class="v-card__text"
        >
          <div>
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Title
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Title"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Short description
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Short description"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Content
            
              </div>
               
              <div
                class="v-input v-textarea v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <textarea
                        placeholder="Short description"
                        rows="5"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <span
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Cancel
        
            </div>
          </button>
           
          <button
            class="v-btn v-btn--loader theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Submit
        
            </div>
            <span
              class="v-btn__loading"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                class="v-progress-circular v-progress-circular--indeterminate"
                role="progressbar"
                style="height: 23px; width: 23px;"
              >
                <svg
                  style="transform: rotate(0deg);"
                  viewBox="21.904761904761905 21.904761904761905 43.80952380952381 43.80952380952381"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    class="v-progress-circular__overlay"
                    cx="43.80952380952381"
                    cy="43.80952380952381"
                    fill="transparent"
                    r="20"
                    stroke-dasharray="125.664"
                    stroke-dashoffset="125.66370614359172px"
                    stroke-width="3.8095238095238093"
                  />
                </svg>
                <div
                  class="v-progress-circular__info"
                />
              </div>
            </span>
          </button>
           
          <div
            class="spacer"
          />
           
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Delete
        
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="flex ml-2 xs12 sm12 md4 xl3"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <!---->
           
          <!---->
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="layout text-xs-center row wrap"
          >
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn v-btn--loader theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Submit
            
                </div>
                <span
                  class="v-btn__loading"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    class="v-progress-circular v-progress-circular--indeterminate"
                    role="progressbar"
                    style="height: 23px; width: 23px;"
                  >
                    <svg
                      style="transform: rotate(0deg);"
                      viewBox="21.904761904761905 21.904761904761905 43.80952380952381 43.80952380952381"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        class="v-progress-circular__overlay"
                        cx="43.80952380952381"
                        cy="43.80952380952381"
                        fill="transparent"
                        r="20"
                        stroke-dasharray="125.664"
                        stroke-dashoffset="125.66370614359172px"
                        stroke-width="3.8095238095238093"
                      />
                    </svg>
                    <div
                      class="v-progress-circular__info"
                    />
                  </div>
                </span>
              </button>
            </div>
             
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn v-btn--disabled theme--light"
                disabled="disabled"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Preview
            
                </div>
              </button>
            </div>
             
            <div
              class="flex xs12"
            >
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`component/doc-templates/BasicInformation Snapshots Populated 1`] = `
<div
  class="container"
>
  <div
    class="layout"
  >
    <div
      class="flex xs12 sm12 md8 xl9"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__title"
        >
          <h3
            class="grey--text"
          >
            
          Basic Information
        
          </h3>
           
          <!---->
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <div
          class="v-card__text"
        >
          <div>
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Title
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder v-input--is-label-active v-input--is-dirty theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Title"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Short description
            
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder v-input--is-label-active v-input--is-dirty theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Short description"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs12"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
              Content
            
              </div>
               
              <div
                class="v-input v-textarea v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder v-input--is-label-active v-input--is-dirty theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <textarea
                        placeholder="Short description"
                        rows="5"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <span
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Cancel
        
            </div>
          </button>
           
          <button
            class="v-btn theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Save changes
        
            </div>
          </button>
           
          <div
            class="spacer"
          />
           
          <button
            class="v-btn theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Delete
        
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="flex ml-2 xs12 sm12 md4 xl3"
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <h3
            class="font-weight-bold mb-2"
          >
            
          Last Modified on
        
          </h3>
           
          <span
            class="mb-2"
          >
            Sep 30, 2019 • 3:51 AM by 
            <u>
              Demo Admin
            </u>
          </span>
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select v-input--is-label-active v-input--is-dirty theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <div
                      class="v-select__selection v-select__selection--comma"
                    >
                      Draft
                    </div>
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          class="primary--text"
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                Draft
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                Published
                              </div>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="layout text-xs-center row wrap"
          >
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Save changes
            
                </div>
              </button>
            </div>
             
            <div
              class="flex mb-2 xs12"
            >
              <button
                class="v-btn v-btn--disabled theme--light"
                disabled="disabled"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
              Preview
            
                </div>
              </button>
            </div>
             
            <div
              class="flex xs12"
            >
              <a
                href="#/move-to-trash"
              >
                Move to trash
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
